lib/iris/analysis/__init__.py

```diff
@@ -1419,36 +1419,42 @@ def interp_order(length):
 """
 
 
-def _build_dask_mdtol_function(dask_nanfunction):
+def _build_dask_mdtol_function(dask_stats_function):
     """
     Make a wrapped dask statistic function that supports the 'mdtol' keyword.
 
-    'dask_nanfunction' must be a statistical function that treats NaNs as
-    missing data, and which has the basic call signature
-    "dask_nanfunction(data, axis, **kwargs)".
+    'dask_function' must be a dask statistical function, compatible with the
+    call signature : "dask_stats_function(data, axis, **kwargs)".
+    It must be masked-data tolerant, i.e. it ignores masked input points and
+    performs a calculation on only the unmasked points.
+    For example, mean([1, --, 2]) = (1 + 2) / 2 = 1.5.
 
     The returned value is a new function operating on dask arrays.
-    It has the call signature "stat(data, axis=-1, mdtol=None, *kwargs)".
+    It has the call signature "stat(data, axis=-1, mdtol=None, **kwargs)".
 
     """
-    @wraps(dask_nanfunction)
+    @wraps(dask_stats_function)
     def inner_stat(array, axis=-1, mdtol=None, **kwargs):
-        dask_result = dask_nanfunction(array, axis=axis, **kwargs)
+        # Call the statistic to get the basic result (missing-data tolerant).
+        dask_result = dask_stats_function(array, axis=axis, **kwargs)
         if mdtol is None:
             result = dask_result
         else:
-            point_counts = np.sum(np.ones(array.shape), axis=axis)
-            point_mask_counts = da.sum(da.isnan(array), axis=axis)
-            masked_point_fractions = (point_mask_counts + 0.0) / point_counts
-            # Note: the +0.0 forces a floating-point divide.
+            # Build a lazy computation to compare the fraction of missing
+            # input points at each output point to the 'mdtol' threshold.
+            point_counts = da.sum(da.ones(array.shape, chunks=array.chunks),
```

Comment:
No, it used not to be but I think it's arguably better if it is.  
Especially if this was quite a large calculation, then this way could chunk it.

As explained in [previous reply to @pelson above](https://github.com/SciTools/iris/pull/2743#discussion_r134248191), there is doubtless a better way of doing this than constructing another large array + collapsing it, but making it a collapse *does* have the advantage of sharing the interpretation of the 'axis' control with the main statistic.